### ISSUE TITLE
ImportError at run_train.sh

### ISSUE THREAD

**Issue by sounansu** [Timen/squeezenext-tensorflow#3]
I try to execute run_train.sh. It occurred error.

```
$ ls
ILSVRC2012_bbox_train_v2.tar.gz  ILSVRC2012_img_train.tar  LICENSE    __init__.py     configs        datasets                          models        predict.py              requirements.txt   run_tensorboard.sh  squeezenext_architecture.py  tabby_cat.jpg          tf-records  train     validation
ILSVRC2012_img_test.tar          ILSVRC2012_img_val.tar    README.md  bounding_boxes  dataloader.py  imagenet_2012_bounding_boxes.csv  optimizer.py  preprocess_imagenet.sh  run_cpu_docker.sh  run_train.sh        squeezenext_model.py         tensorflow_extentions  tools       train.py
$ cat run_train.sh
#!/usr/bin/env bash
DATE=`date '+%Y-%m-%d_%H-%M'`
#TRAIN_PATH="/usr/local/share/models/"
TRAIN_PATH="/home/sounansu/anaconda3/squeezenext-tensorflow/models/"
TRAIN_DIR=$TRAIN_PATH$DATE

if [[ ! -e $DATA_DIR ]]; then
    echo "Data dir $DATA_DIR does not exists." 1>&2
    exit 1
fi
if [[ ! -e $TRAIN_DIR ]]; then
    mkdir $TRAIN_DIR
elif [[ ! -d $TRAIN_DIR ]]; then
    echo "Model dir $TRAIN_DIR already exists but is not a directory" 1>&2
fi

PYTHONPATH="./" python train.py \
--model_dir $TRAIN_DIR \
--configuration "v_1_0_SqNxt_23" \
--batch_size 256 \
--num_epochs 120 \
--training_file_pattern $DATA_DIR"train-*" \
--validation_file_pattern $DATA_DIR"validation-*"
$ bash run_train.sh
/home/sounansu/anaconda3/envs/squeezenext-tensorflow/lib/python3.5/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Traceback (most recent call last):
  File "train.py", line 3, in <module>
    from configs import configs
  File "/home/sounansu/anaconda3/squeezenext-tensorflow/configs/__init__.py", line 6, in <module>
    configs[module[:-3]] = __import__(module[:-3], locals(), globals()).training_params
ImportError: No module named 'v_1_0_G_SqNxt_23'
```

Please gime me some advices!

**Comment by sounansu** [comment]
I change python3 to python2. 
There was no errors at Import time.

But, another errors are occured.

```
WARNING:tensorflow:No new checkpoint ready for evaluation. Skip the current evaluation pass as evaluation results are expected to be same for the same checkpoint.
Traceback (most recent call last):
  File "train.py", line 88, in <module>
    tf.app.run(main)
  File "/home/sounansu/anaconda3/envs/squeezenext-tensorflow/lib/python2.7/site-packages/tensorflow/python/platform/app.py", line 126, in run
    _sys.exit(main(argv))
  File "train.py", line 78, in main
    tf.estimator.train_and_evaluate(classifier, train_spec, eval_spec)
  File "/home/sounansu/anaconda3/envs/squeezenext-tensorflow/lib/python2.7/site-packages/tensorflow/python/estimator/training.py", line 439, in train_and_evaluate
    executor.run()
  File "/home/sounansu/anaconda3/envs/squeezenext-tensorflow/lib/python2.7/site-packages/tensorflow/python/estimator/training.py", line 518, in run
    self.run_local()
  File "/home/sounansu/anaconda3/envs/squeezenext-tensorflow/lib/python2.7/site-packages/tensorflow/python/estimator/training.py", line 663, in run_local
    'Eval status: {}'.format(eval_result.status))
RuntimeError: There was no new checkpoint after the training. Eval status: no new checkpoint
```
I want another advices!

**Comment by Timen** [comment]
Could you maybe post the full output in the console so I may have an idea at which point in the process the error occurs? I updated the readme to reflect that the code was created for python 2.7.

**Comment by sounansu** [comment]
@Timen ! thank you for your reply.

I change `run_train.sh` for my directory.
```
$ cat run_train.sh
#!/usr/bin/env bash
DATE=`date '+%Y-%m-%d_%H-%M'`
#TRAIN_PATH="/usr/local/share/models/"
TRAIN_PATH="/home/sounansu/anaconda3/squeezenext-tensorflow/models/"
TRAIN_DIR=$TRAIN_PATH$DATE

if [[ ! -e $DATA_DIR ]]; then
    echo "Data dir $DATA_DIR does not exists." 1>&2
    exit 1
fi
if [[ ! -e $TRAIN_DIR ]]; then
    mkdir $TRAIN_DIR
elif [[ ! -d $TRAIN_DIR ]]; then
    echo "Model dir $TRAIN_DIR already exists but is not a directory" 1>&2
fi

PYTHONPATH="./" python train.py \
--model_dir $TRAIN_DIR \
--configuration "v_1_0_SqNxt_23" \
--batch_size 256 \
--num_epochs 120 \
--training_file_pattern $DATA_DIR"train-*" \
--validation_file_pattern $DATA_DIR"validation-*"
```

and the full output are below
```
$ export DATA_DIR=./;bash run_train.sh
INFO:tensorflow:Using default config.
INFO:tensorflow:Using config: {'_save_checkpoints_secs': 600, '_session_config': None, '_keep_checkpoint_max': 5, '_task_type': 'worker', '_train_distribute': None, '_is_chief': True, '_cluster_spec': <tensorflow.python.training.server_lib.ClusterSpec object at 0x7f989e5201d0>, '_evaluation_master': '', '_save_checkpoints_steps': None, '_keep_checkpoint_every_n_hours': 10000, '_service': None, '_num_ps_replicas': 0, '_tf_random_seed': None, '_master': '', '_num_worker_replicas': 1, '_task_id': 0, '_log_step_count_steps': 100, '_model_dir': '/home/sounansu/anaconda3/squeezenext-tensorflow/models/2018-11-01_20-36', '_global_id_in_cluster': 0, '_save_summary_steps': 100}
INFO:tensorflow:Total steps = 600480, num_epochs = 120, batch size = 256
INFO:tensorflow:Running training and evaluation locally (non-distributed).
INFO:tensorflow:Start train and evaluate loop. The evaluate will happen after 1800 secs (eval_spec.throttle_secs) or training is finished.
INFO:tensorflow:Calling model_fn.
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Create CheckpointSaverHook.
INFO:tensorflow:Graph was finalized.
2018-11-01 20:36:44.617115: I tensorflow/core/platform/cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2018-11-01 20:36:44.726167: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:898] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2018-11-01 20:36:44.726652: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1356] Found device 0 with properties:
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.6705
pciBusID: 0000:1d:00.0
totalMemory: 10.91GiB freeMemory: 7.35GiB
2018-11-01 20:36:44.726667: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1435] Adding visible gpu devices: 0
2018-11-01 20:36:45.091236: I tensorflow/core/common_runtime/gpu/gpu_device.cc:923] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-11-01 20:36:45.091273: I tensorflow/core/common_runtime/gpu/gpu_device.cc:929]      0
2018-11-01 20:36:45.091279: I tensorflow/core/common_runtime/gpu/gpu_device.cc:942] 0:   N
2018-11-01 20:36:45.091465: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1053] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 7100 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:1d:00.0, compute capability: 6.1)
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Saving checkpoints for 0 into /home/sounansu/anaconda3/squeezenext-tensorflow/models/2018-11-01_20-36/model.ckpt.
INFO:tensorflow:Loss for final step: None.
INFO:tensorflow:Calling model_fn.
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Starting evaluation at 2018-11-01-11:36:59
INFO:tensorflow:Graph was finalized.
2018-11-01 20:37:01.836180: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1435] Adding visible gpu devices: 0
2018-11-01 20:37:01.836220: I tensorflow/core/common_runtime/gpu/gpu_device.cc:923] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-11-01 20:37:01.836227: I tensorflow/core/common_runtime/gpu/gpu_device.cc:929]      0
2018-11-01 20:37:01.836237: I tensorflow/core/common_runtime/gpu/gpu_device.cc:942] 0:   N
2018-11-01 20:37:01.836348: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1053] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 7100 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:1d:00.0, compute capability: 6.1)
INFO:tensorflow:Restoring parameters from /home/sounansu/anaconda3/squeezenext-tensorflow/models/2018-11-01_20-36/model.ckpt-0
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Finished evaluation at 2018-11-01-11:37:03
INFO:tensorflow:Saving dict for global step 0: Recall@1 = 0.0, Recall@5 = nan, global_step = 0, loss = 0.0
INFO:tensorflow:Calling model_fn.
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Create CheckpointSaverHook.
INFO:tensorflow:Graph was finalized.
2018-11-01 20:37:11.220277: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1435] Adding visible gpu devices: 0
2018-11-01 20:37:11.220318: I tensorflow/core/common_runtime/gpu/gpu_device.cc:923] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-11-01 20:37:11.220325: I tensorflow/core/common_runtime/gpu/gpu_device.cc:929]      0
2018-11-01 20:37:11.220331: I tensorflow/core/common_runtime/gpu/gpu_device.cc:942] 0:   N
2018-11-01 20:37:11.220474: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1053] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 7100 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:1d:00.0, compute capability: 6.1)
INFO:tensorflow:Restoring parameters from /home/sounansu/anaconda3/squeezenext-tensorflow/models/2018-11-01_20-36/model.ckpt-0
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Saving checkpoints for 0 into /home/sounansu/anaconda3/squeezenext-tensorflow/models/2018-11-01_20-36/model.ckpt.
INFO:tensorflow:Loss for final step: None.
WARNING:tensorflow:No new checkpoint ready for evaluation. Skip the current evaluation pass as evaluation results are expected to be same for the same checkpoint.
Traceback (most recent call last):
  File "train.py", line 88, in <module>
    tf.app.run(main)
  File "/home/sounansu/anaconda3/envs/squeezenext-tensorflow/lib/python2.7/site-packages/tensorflow/python/platform/app.py", line 126, in run
    _sys.exit(main(argv))
  File "train.py", line 78, in main
    tf.estimator.train_and_evaluate(classifier, train_spec, eval_spec)
  File "/home/sounansu/anaconda3/envs/squeezenext-tensorflow/lib/python2.7/site-packages/tensorflow/python/estimator/training.py", line 439, in train_and_evaluate
    executor.run()
  File "/home/sounansu/anaconda3/envs/squeezenext-tensorflow/lib/python2.7/site-packages/tensorflow/python/estimator/training.py", line 518, in run
    self.run_local()
  File "/home/sounansu/anaconda3/envs/squeezenext-tensorflow/lib/python2.7/site-packages/tensorflow/python/estimator/training.py", line 663, in run_local
    'Eval status: {}'.format(eval_result.status))
RuntimeError: There was no new checkpoint after the training. Eval status: no new checkpoint
```

**Comment by Timen** [comment]
What version of tensorflow are you using?

What kind of data are you using? Did you create the dataset according to the instructions here or did you create your own tf records?

**Comment by sounansu** [comment]
@Timen  Thank you! My answers are below.

> What version of tensorflow are you using?
```
$ python -c 'import tensorflow as tf; print(tf.__version__)'
1.8.0
```
I prepared datas as below.
I get `ILSVRC2012_img_test.tar ILSVRC2012_img_train.tar ILSVRC2012_img_val.tar ILSVRC2012_bbox_train_v2.tar.gz` and copy to `squeezenext-tensorflow` directory
and
```
export DATA_DIR=./
bash datasets/process_downloaded_imagenet.sh $DATA_DIR
```
below list are after execute command
```
$ ls
ILSVRC2012_bbox_train_v2.tar.gz   requirements.txt
ILSVRC2012_img_test.tar           run_cpu_docker.sh
ILSVRC2012_img_train.tar          run_tensorboard.sh
ILSVRC2012_img_val.tar            run_train.sh
LICENSE                           squeezenext
README.md                         squeezenext.zip
__init__.py                       squeezenext_architecture.py
bounding_boxes                    squeezenext_architecture.pyc
configs                           squeezenext_model.py
dataloader.py                     squeezenext_model.pyc
dataloader.pyc                    tabby_cat.jpg
datasets                          tensorflow_extentions
imagenet_2012_bounding_boxes.csv  tf-records
models                            tmp.log
optimizer.py                      tools
optimizer.pyc                     train
predict.py                        train.py
preprocess_imagenet.sh            validation
```
and
`bash run_train.sh`

**Comment by Timen** [comment]
Ok I see you set you DATA_DIR to the same directory as your code directory. I didn't do it like this, and it is also smarter to have the imagenet data in a different directory than the code (keeps it nicely seperated).

However it might still work fine, are there tf-records inside the tf-records folder? And is the size of the tf-records folder pretty big think around 100GB if i remember correctly?

**Comment by Timen** [comment]
If the data is in the tf-records folder I think I made a mistake and I'll update the code but you first need to make sure the tf-records were generated.

**Comment by sounansu** [comment]
@Timen  Thanks quickly response.

```
$ du -s *
20376   ILSVRC2012_bbox_train_v2.tar.gz
13365056        ILSVRC2012_img_test.tar
144431156       ILSVRC2012_img_train.tar
6586844 ILSVRC2012_img_val.tar
4       LICENSE
8       README.md
0       __init__.py
2202748 bounding_boxes
24276   configs
8       dataloader.py
8       dataloader.pyc
1328    datasets
29016   imagenet_2012_bounding_boxes.csv
85300   models
4       optimizer.py
4       optimizer.pyc
4       predict.py
4       preprocess_imagenet.sh
4       requirements.txt
4       run_cpu_docker.sh
4       run_tensorboard.sh
4       run_train.sh
52612   squeezenext
5808    squeezenext.zip
8       squeezenext_architecture.py
8       squeezenext_architecture.pyc
8       squeezenext_model.py
8       squeezenext_model.pyc
988     tabby_cat.jpg
32      tensorflow_extentions
150653544       tf-records
0       tmp.log
44      tools
290518576       train
4       train.py
6655764 validation
$ ls tf-records |wc
   1152    1152   24832
```
Are there correct?

If it is correct, another member of our team will try another machine.
(So, if HDD  has enough space.......)

Thank you!

**Comment by Timen** [comment]
Allright, then I made an error, I just updated the repo so if you pull the repo and try to run run_train.sh it should work.

**Comment by sounansu** [comment]
@Timen !  Yes Yes!
I pull your new code. and I can run `run_train.sh` with NO ERROR.

Very very thanks.

```
$ bash run_train.sh
INFO:tensorflow:Using default config.
INFO:tensorflow:Using config: {'_save_checkpoints_secs': 600, '_session_config': None, '_keep_checkpoint_max': 5, '_task_type': 'worker', '_train_distribute': None, '_is_chief': True, '_cluster_spec': <tensorflow.python.training.server_lib.ClusterSpec object at 0x7f9f44320210>, '_evaluation_master': '', '_save_checkpoints_steps': None, '_keep_checkpoint_every_n_hours': 10000, '_service': None, '_num_ps_replicas': 0, '_tf_random_seed': None, '_master': '', '_num_worker_replicas': 1, '_task_id': 0, '_log_step_count_steps': 100, '_model_dir': '/home/sounansu/anaconda3/squeezenext-tensorflow/models/2018-11-05_18-01', '_global_id_in_cluster': 0, '_save_summary_steps': 100}
INFO:tensorflow:Total steps = 600480, num_epochs = 120, batch size = 256
INFO:tensorflow:Running training and evaluation locally (non-distributed).
INFO:tensorflow:Start train and evaluate loop. The evaluate will happen after 1800 secs (eval_spec.throttle_secs) or training is finished.
INFO:tensorflow:Calling model_fn.
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Create CheckpointSaverHook.
INFO:tensorflow:Graph was finalized.
2018-11-05 18:01:10.667498: I tensorflow/core/platform/cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2018-11-05 18:01:10.778377: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:898] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2018-11-05 18:01:10.778907: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1356] Found device 0 with properties:
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.6705
pciBusID: 0000:1d:00.0
totalMemory: 10.91GiB freeMemory: 10.76GiB
2018-11-05 18:01:10.778923: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1435] Adding visible gpu devices: 0
2018-11-05 18:01:10.943665: I tensorflow/core/common_runtime/gpu/gpu_device.cc:923] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-11-05 18:01:10.943701: I tensorflow/core/common_runtime/gpu/gpu_device.cc:929]      0
2018-11-05 18:01:10.943707: I tensorflow/core/common_runtime/gpu/gpu_device.cc:942] 0:   N
2018-11-05 18:01:10.943918: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1053] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10415 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:1d:00.0, compute capability: 6.1)
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
2018-11-05 18:01:30.749034: I tensorflow/core/kernels/data/shuffle_dataset_op.cc:94] Filling up shuffle buffer (this may take a while): 4114 of 8192
2018-11-05 18:01:40.194971: I tensorflow/core/kernels/data/shuffle_dataset_op.cc:129] Shuffle buffer filled.
INFO:tensorflow:Saving checkpoints for 1 into /home/sounansu/anaconda3/squeezenext-tensorflow/models/2018-11-05_18-01/model.ckpt.
INFO:tensorflow:loss = 6.921589, step = 0
INFO:tensorflow:global_step/sec: 1.70169
INFO:tensorflow:loss = 6.8939543, step = 100 (58.765 sec)
INFO:tensorflow:global_step/sec: 1.72546
INFO:tensorflow:loss = 6.870297, step = 200 (57.956 sec)
INFO:tensorflow:global_step/sec: 1.99625
INFO:tensorflow:loss = 6.8319755, step = 300 (50.094 sec)
INFO:tensorflow:global_step/sec: 1.7547
INFO:tensorflow:loss = 6.8553343, step = 400 (56.989 sec)


```

**Comment by Timen** [comment]
Great! I'll quickly explain what I did wrong, first of all I didn't fully test the setup process on a clean system so thank you for testing it for me. Second tensorflow estimators are notorious for outputting very little debug data but in my experience when everything runs without errors but nothing happens something is wrong with the input data. 

In this case it was looking for the train and validation tf-records in the DATA_DIR, however I changed the process_download_imagenet script to output those to $DATA_DIR/tf-records/. So that is all I did i updated the run_train script to look for the tf-records in the  $DATA_DIR/tf-records/ folder instead of $DATA_DIR.
